radixdlt/babylon-wallet-android
# [ABW-3097] Add Google Sign-In and authorization flows

## Description
This PR adds:
- dependencies for Google Sign-In and Drive API
- authentication and authorization flows for enabling new cloud backup solution. Authentication means to sign in with email. Authorization means to give consent for Drive API
- a new screen to inform the user about cloud back up solution when creating wallet for first time

The flows are based on this [diagram](https://whimsical.com/android-cloud-backup-solution-opt-in-XEJA7GNmDAUB3mygwaFycU). Please take a look at it before you ask questions. 

- This PR **does not implement the new backup solution,** and the old one is still in place. 
- This PR replaces the logic under the switch of automatic backups in Backup settings screen with the sign in flow. 
- This PR will not get merged and will be the base for the whole feature. The next PR will introduce the Drive client to sync the profile.

⚠️   The cloud project has been created with my personal gmail. 

## How to test

1. use one of the above flows to authenticate and authorize with your email
2. in your desktop browser navigate to the settings of your Google Drive and then to the managed apps
3. if you see the `ioannidis-auth` then it's a successful auth 

## Video

https://github.com/radixdlt/babylon-wallet-android/assets/118305718/4aff2f32-d29d-4ab7-86e6-a3a50e7a79e7


## PR submission checklist
- [X] I have tested all sign in flows
